src/main/java/bio/terra/service/resourcemanagement/google/GoogleProjectService.java

```diff
@@ -197,6 +202,16 @@ private GoogleProjectResource newProject(
         }
     }
 
+    @VisibleForTesting
+    static void validateProjectId(final String projectId) {
+        Preconditions.checkArgument(
+            projectId.matches("[a-z][a-z0-9-]{5,29}(?<!-)$"),
```

Comment:
Is this the actual length? Should it be {6, 30}?

Reply:
Plus, this case passes - exactly 30 letters: GoogleProjectService.validateProjectId("a01234567890123456789012345678");

Reply:
Haha yeah, it's a little confusing.  The first `[a-z]` takes a character away, that's why I "minused 1" in the `{5,29}` bit

Reply:
It's one less because of the first character set [a-z] at the start. So that's the missing fist char

Reply:
Ah, gotcha! Thanks for clarifying.